cobigen/cobigen-core-parent/cobigen-core/src/main/java/com/devonfw/cobigen/impl/config/reader/ContextConfigurationReader.java

```diff
@@ -54,9 +57,23 @@
      */
     public ContextConfigurationReader(Path configRoot) throws InvalidConfigurationException {
 
-        Path contextFile = configRoot.resolve(ConfigurationConstants.CONTEXT_CONFIG_FILENAME);
-        String filePath = contextFile.toAbsolutePath().toString();
+        contextFile = configRoot.resolve(ConfigurationConstants.CONTEXT_CONFIG_FILENAME);
+        if (!Files.exists(contextFile)) {
+            configRoot = configRoot.resolve(ConfigurationConstants.TEMPLATE_RESOURCE_FOLDER);
+            contextFile = configRoot.resolve(ConfigurationConstants.CONTEXT_CONFIG_FILENAME);
+            if (!Files.exists(contextFile)) {
+                throw new InvalidConfigurationException(contextFile, "Could not read context configuration file.");
```

Comment:
better: Could not find configuration file as read would imply that you have found it, but you are unable to read it.